Mathlib/Order/Bounds/Basic.lean

```diff
@@ -1636,6 +1636,41 @@ theorem isGLB_prod [Preorder α] [Preorder β] {s : Set (α × β)} (p : α × 
   @isLUB_prod αᵒᵈ βᵒᵈ _ _ _ _
 #align is_glb_prod isGLB_prod
 
+lemma BddAbove.range_mono [Preorder β] {f : α → β} (g : α → β) (h : ∀ a, f a ≤ g a)
+    (hbdd : BddAbove (Set.range g)) : BddAbove (Set.range f) := by
+  rw [bddAbove_def] at hbdd ⊢
+  obtain ⟨C, hC⟩ := hbdd
+  refine ⟨C, fun y hy => ?_⟩
+  rw [Set.mem_range] at hy
+  obtain ⟨y', hy'⟩ := hy
+  calc y = f y'  := hy'.symm
+       _ ≤ g y'  := h _
+       _ ≤ C := hC (g y') (by simp)
+
+lemma BddBelow.range_mono [Preorder β] (f : α → β) {g : α → β} (h : ∀ a, f a ≤ g a)
+    (hbdd : BddBelow (Set.range f)) : BddBelow (Set.range g) :=
+  BddAbove.range_mono (β := OrderDual β) f h hbdd
+
+lemma BddAbove.range_comp {γ : Type*} [Preorder β] [Preorder γ] {f : α → β} {g : β → γ}
+    (hf : BddAbove (Set.range f)) (hg : Monotone g) : BddAbove (Set.range (fun x => g (f x))) := by
+  rw [bddAbove_def] at hf ⊢
+  obtain ⟨C, hC⟩ := hf
+  refine ⟨g C, fun y hy => ?_⟩
+  rw [Set.mem_range] at hy
+  obtain ⟨y', hy'⟩ := hy
+  calc y = g (f y') := hy'.symm
+      _ ≤ g C := hg <| hC (f y') <| Set.mem_range_self _
+
+lemma BddBelow.range_comp {γ : Type*} [Preorder β] [Preorder γ] {f : α → β} {g : β → γ}
+    (hf : BddBelow (Set.range f)) (hg : Monotone g) : BddBelow (Set.range (fun x => g (f x))) := by
+  rw [bddBelow_def] at hf ⊢
+  obtain ⟨C, hC⟩ := hf
+  refine ⟨g C, fun y hy => ?_⟩
+  rw [Set.mem_range] at hy
+  obtain ⟨y', hy'⟩ := hy
+  calc y = g (f y') := hy'.symm
+      _ ≥ g C := hg <| hC (f y') <| Set.mem_range_self _
```

Comment:
Can't this also be golfed using `OrderDual`?

Reply:
Not easily. We need to take the dual of both `β` and `γ`, and I think the `Monotone g` hypothesis needs to be adapted. Maybe it's possible but it's easier to copy/paste.